src/app/api/notification/notification.service.ts

```diff
@@ -49,16 +49,71 @@ export class NotificationService extends BaseService {
             email: getEmailDetails(actionUserName, task.title)[action],
           },
         }
-
         return copilotUtils.createNotification(notificationDetails)
       })
 
-      await Promise.all(promises)
+      return await Promise.all(promises)
     } catch (error) {
       console.error(`Failed to send notifications for action: ${action}`, error)
     }
   }
 
+  /**
+   * Adds a ClientNotification column associated with a task
+   * @param task Associated task
+   * @param notification Associated notification
+   * @returns New ClientNotification object
+   */
+  async addToClientNotifications(task: Task, notification: NotificationCreatedResponse): Promise<ClientNotification> {
+    return await this.db.clientNotification.create({
+      data: {
+        clientId: z.string().parse(notification.recipientId),
+        notificationId: notification.id,
+        taskId: task.id,
+      },
+    })
+  }
+
+  /**
+   * Marks a client notification in Copilot Notifications service as read
+   * @param id Notification ID for object as exists in Copilot
+   */
+  async markClientNotificationAsRead(task: Task) {
+    const copilot = new CopilotAPI(this.user.token)
+    try {
+      const relatedNotification = await this.db.clientNotification.findFirst({
+        where: {
+          clientId: z.string().parse(task.assigneeId),
+          taskId: task.id,
+        },
+      })
+      if (!relatedNotification) {
+        console.error(`Failed to delete client notification for task id ${task.id}`)
```

Comment:
```suggestion
        console.error(`Failed to delete client notification for task id ${task.id} because the notification for client ${task.assigneeId} was not found`)
```